iothub/device/src/Common/Amqp/ClientWebSocketTransport.cs

```diff
@@ -166,24 +162,24 @@ private async Task<int> ReadAsyncCore(TransportAsyncCallbackArgs args)
             {
                 if (!succeeded)
                 {
-                    this.Abort();
+                    Abort();
                 }
             }
         }
 
         protected override bool OpenInternal()
         {
-            this.ThrowIfNotOpen();
+            ThrowIfNotOpen();
 
             return true;
         }
 
         protected override bool CloseInternal()
         {
-            var webSocketState = this.webSocket.State;
+            WebSocketState webSocketState = _webSocket.State;
             if (webSocketState != WebSocketState.Closed && webSocketState != WebSocketState.Aborted)
             {
-                this.CloseInternalAsync(CloseTimeout);
+                CloseInternalAsync(s_closeTimeout).GetAwaiter().GetResult();
```

Comment:
We weren't awaiting the completion of the task here!

Reply:
Crazy, right?